passes/opt/opt_clean.cc

```diff
@@ -31,6 +31,156 @@ PRIVATE_NAMESPACE_BEGIN
 
 using RTLIL::id2cstr;
 
+bool should_be_kept(Design *d, Cell *cell)
```

Comment:
Before we merge (if we merge), we should get rid of this `should_be_kept` function and use `keep_cache` and the other code that's already there.